patroni/ha.py

```diff
@@ -304,20 +317,52 @@ def is_healthiest_node(self):
         members = {m.name: m for m in self.cluster.members + self.old_cluster.members}
         return self._is_healthiest_node(members.values())
 
+    def release_leader_key_voluntarily(self):
+        self.dcs.delete_leader()
+        self.touch_member()
+        self.dcs.reset_cluster()
+        logger.info("Leader key released")
+
     def demote(self, delete_leader=True):
         if delete_leader:
             self.state_handler.stop()
             self.state_handler.set_role('demoted')
-            self.dcs.delete_leader()
-            self.touch_member()
-            self.dcs.reset_cluster()
+            self.release_leader_key_voluntarily()
             sleep(2)  # Give a time to somebody to take the leader lock
             cluster = self.dcs.get_cluster()
             node_to_follow = self._get_node_to_follow(cluster)
             self.state_handler.follow(node_to_follow, cluster.leader, recovery=True, need_rewind=True)
         else:
             self.state_handler.follow(None, None)
 
+    def demote_asap(self):
+        """We are master, but we want to relinquish control ASAP.
+
+        We don't bother to shut down properly because we will need to rewind anyway."""
+        self.state_handler.stop('immediate', checkpoint=False)
+        # Don't set role to unknown to ensure we get rewound
```

Comment:
Role should be set to 'demoted', otherwise it is still reported as 'master' via API, what stops other nodes from promote, because for them the old master is a kind of still alive.

Setting role to 'demote' will not break `pg_rewind`.